tests/console/yast2_lan.pm

```diff
@@ -34,18 +34,71 @@ sub handle_dhcp_popup {
     }
 }
 
+
+sub check_etc_hosts_update {
+    my $looprun = 1;
+    assert_script_run "cat /etc/hosts";
+    until ($looprun == 3) {
+        script_run("yast2 lan; echo yast2-lan-status-\$? > /dev/$serialdev", 0);
+        assert_screen "yast2_lan";
+        send_key 'alt-i';    # edit NIC
+        assert_screen 'yast_ncurses_network_card_setup';
+        send_key 'alt-t';    # set to static ip
+        assert_screen 'yast_ncurses_set_static_ip';
+        send_key 'tab';
+        for (1 .. 15) { send_key "backspace" };    # delete existing IP if any
+        type_string "192.168.122.10";
+        send_key 'tab';
+        for (1 .. 15) { send_key "backspace" };    # delete existing netmask if any
+        type_string "/24";
+        send_key 'tab';
+        for (1 .. 50) { send_key "backspace" }
+        type_string "susetest.test$looprun";
+        assert_screen 'yast_ncurses_static_ip_set';
+        send_key 'alt-n';
+        wait_still_screen;
+        send_key 'alt-o';
+        wait_still_screen;
+        wait_serial("yast2-lan-status-0", 180) || die "'yast2 lan' didn't finish";
+        if ($looprun < 3) {
```

Comment:
you inside a loop with condition `until($looprun == 3) ` so why you need this if here ?

Reply:
Hello, you are right regarding the if, it survived from the first version of the script. I removed it.